open-api/rest-catalog-open-api.yaml

```diff
@@ -3324,6 +3348,97 @@ components:
           type: integer
           format: int64
 
+    TypeValue:
+      oneOf:
+        - $ref: '#/components/schemas/PrimitiveTypeValue'
```

Comment:
My take is that we should have one for each type, for example `#/components/schemas/DecimalTypeValue` and `#/components/schemas/TimestampTypeValue` although they are all string, and then we can document how each type is serialized differently in the definition section of each type value.

Reply:
thoughts? @danielcweeks @nastra @amogh-jahagirdar

Reply:
I think documenting how the different types are serialized is important but just to make sure for `DecimalTypeValue` and `TimestampTypeValue` those are just examples right? We don't need those data types spec'd out for what we're trying to do here right?

Reply:
So to summarize the 2 options we are discussing so far:

Option 1: different schema

```
TypeValue:
  oneOf:
    - $ref: '#/components/schemas/DecimalTypeValue'
    - $ref: '#/components/schemas/TimestampTypeValue'
    - ...
 
DecimalTypeValue:
  type: string
  description: // describe serialization for decimal
  ...

TimestampTypeValue:
  type: string
  description: // describe serialization for timestamp
  ...

```

Option 2: same schema, just document different serializations

```
TypeValue:
      oneOf:
        - $ref: '#/components/schemas/PrimitiveTypeValue'
        - ...
 
PrimitiveTypeValue:
      description: // describe serialization for all primitive types
      oneOf:
        - string
        - ...
```

@rdblue @danielcweeks @amogh-jahagirdar @rahil-c

Reply:
Per the conversation in the prior PR about passing content files to the service The first approach beneficial if there is a use case for building custom deserializers

Reply:
Ah my bad I only saw ContentFile, for the actual update we need schema/partition-spec so yes these data types do need to be spec'd out in rest.

Option 1 seems better to me; it does not have as much indirection as option 2 has and the types are just explicitly laid out.

Reply:
yeah +1 for option 1. I would suggest @geruh let's first move in that direction and make the corresponding changes, while waiting for more feedback.

Reply:
+1 for option 1.

I also [commented](https://github.com/apache/iceberg/pull/9695#pullrequestreview-1889027020) on #9695 about the non-primitive types. I think that structs should use a simple array of these primitive values.

@jackye1995, should we move these additions to a separate PR so that the scan APIs are not blocked by the append API?

Reply:
The append API change is pretty minimal if we exclude these type changes, that's why it is kept here. But seems like we have other thoughts about how the append API as a separated API. In that case +1 let's separate these 2 changes. 

We can either create a new PR for the Append API, or the content file. @geruh up to you.

Reply:
I changed the focus of this PR to the `ContentFile` and type spec changes, to preserve the comments. I'll open a new PR later for the Append changes.